runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -4106,7 +4106,9 @@ static void jitHookClassLoad(J9HookInterface * * hookInterface, UDATA eventNum,
          allocFailed = true;
       }
 
-   compInfo->getPersistentInfo()->ensureUnloadedAddressSetsAreInitialized();
+   // In JITaaS server mode unloaded addresses are maintained per client
+   if (compInfo->getPersistentInfo()->getJITaaSMode() != SERVER_MODE)
```

Comment:
I checked that the hook was being called on the server and it is unfortunately, presumably because bootstrap classes are still loaded. There are various other JITaaS mode checks in that code so it seems to have been the case all along at least.

Anyway I've moved the check into `ensureUnloadedAddressSetsAreInitialized()` instead to make sure the single persistent instances are never created on the server regardless of where it's called from.